sklearn/utils/estimator_checks.py

```diff
@@ -452,7 +454,15 @@ def check_estimator(Estimator, generate_only=False):
     if generate_only:
         return checks_generator
 
+    xfail_checks = _safe_tags(estimator, '_xfail_test')
+
     for estimator, check in checks_generator:
+        check_name = _set_check_estimator_ids(check)
+        if xfail_checks and check_name in xfail_checks:
+            # skip tests marked as a known failure and raise a warning
+            msg = xfail_checks[check_name]
+            warnings.warn(f'Skipping {check_name}: {msg}', SkipTestWarning)
+            continue
```

Comment:
Can we apply the xfail decorator as a function instead of manually replicating its behavior?

I.e.

check = pytest.mark.xfail(check)

or something like that? 

Because we have the try / except logic just below (and we might want to update the comment about pandas)

Reply:
We can't use pytest in this file, since it's supposed to work without it.

There is indeed some slight redundancy with _mark_xfail_checks but I think trying to factorize it might be more trouble than what it's worth. It's just 4 extra lines in the end.

Updated the comment about pandas.

Reply:
I'd say the introduction of the tag is a good reason to start requiring `pytest` for using `check_estimator` now. But agreed that's another story.


>  It's just 6 extra lines in the end

Sure, though I find our whole test suite really hard to maintain in general.

Reply:
> Sure, though I find our whole test suite really hard to maintain in general.

Agreed, I'm just saying that slightly more verbosity and 2 repetitions is easier to maintain  than coming up with some function wrappers. The problem is not so much lines of code as complexity.